documentation/src/docs/streamCuts.md

```diff
@@ -0,0 +1,87 @@
+<!--
+Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+-->
+# Working with Pravega: StreamCuts
+
+This section describes about StreamCuts and how it can be used with streaming clients and batch clients.
+Pre-requisites: You should be familiar with Pravega Concepts (see Pravega Concepts).
+
+## Definition
+
+StreamCut represents a consistent position in the stream. It essentially contains a set of segment
+and offset pairs for a single stream which represents the complete keyspace. The offset always points to
+the event boundary and hence there will be no offset pointing to an incomplete event.
+
+Given that data can be continuously added to a stream, the streamcut representing the tail of the
+stream(with the newest event) is an ever changing one. Similarly the stream cut representing the
+head of the stream(with the oldest event) is an ever changing one as the stream retention policy
+could truncate the stream. ```StreamCut.UNBOUNDED``` is used to represent such a position in the stream and
+the user can use it to specify this ever changing stream position (both head and tail of the stream).
+
+It should be noted that streamcuts obtained using the streaming client and batch client can be used
+interchangeably.
+
+## StreamCut with Streaming clients.
+
+StreamCut(s) can be obtained from a Reader group using the following api.
```

Comment:
While this is true, I also think that here there are too many concepts hidden for a first-time reader. I would add a little bit more of introductory lines to make this self-contained. Concretely, it would be nice to mention the relationship between a ReaderGroup and its Readers. Even more important, the idea hidden here is that a call `getStreamCuts()` provides the streamcut representing the "current point" in which readers are positioned. Therefore, there is an inherent relationship between time and StreamCuts that would require a mention as well. This is important to clarify for a developer trying to understand these concepts and use the API in a code example.

Reply:
updated it with the points suggested.